graph/ui-project-details/src/lib/project-details/project-details.tsx

```diff
@@ -28,25 +31,44 @@ export interface ProjectDetailsProps {
   viewInProjectGraphPosition?: 'top' | 'bottom';
 }
 
-const typeToProjectType = {
-  app: 'Application',
-  lib: 'Library',
-  e2e: 'E2E',
-};
+export interface SingleProjectDetailsProps extends ProjectDetailsPropsBase {
```

Comment:
I think a discriminated union type would be cleaner?

```ts
export type SingleProjectDetailsProps = ProjectDetailsPropsBase & {
  type: 'single';
  project: ProjectGraphProjectNode;
  sourceMap: Record<string, string[]>;
};

export type MultiProjectDetailsProps = ProjectDetailsPropsBase & {
  type: 'multi';
  projects: ProjectGraphProjectNode[];
  sourceMaps: Record<string, Record<string, string[]>>;
};

function getSingleProjectProps(
  props: SingleProjectDetailsProps | MultiProjectDetailsProps,
  project: ProjectGraphProjectNode
): SingleProjectDetailsProps {
  return props.type === 'single'
    ? props
    : {
      ...props,
      type: 'single',
      project,
      sourceMap: props.sourceMaps[project.data.root],
    };
}

```

Reply:
it looks nicer and I updated it to be types but I don't like having to pass a `type` property since it affects all the places where the component is used. I'd just stick with inferring the type from `project` vs `projects`